onnxruntime/contrib_ops/cpu/signal/dft.cc

```diff
@@ -242,60 +262,58 @@ static Status dft_naive(const Tensor* X, Tensor* Y,
   // Get data
   auto* X_data = const_cast<U*>(reinterpret_cast<const U*>(X->DataRaw())) + X_offset;
   auto* Y_data = reinterpret_cast<std::complex<T>*>(Y->MutableDataRaw()) + Y_offset;
-  
+
   U* window_data = nullptr;
   if (window) {
     window_data = const_cast<U*>(reinterpret_cast<const U*>(window->DataRaw()));
   }
 
-  auto angular_velocity = compute_angular_velocity<T>(number_of_samples, inverse);
+  auto angular_velocity = compute_angular_velocity<T>(dft_length, inverse);
 
   for (size_t i = 0; i < dft_output_size; i++) {
     std::complex<T>& out = *(Y_data + i*Y_stride);
     out.real(0);
     out.imag(0);
 
-    for (size_t j = 0; j < number_of_samples; j++) {  // vectorize over this loop
+    for (size_t j = 0; j < dft_length; j++) {  // vectorize over this loop
       auto exponential = std::complex<T>(cos(i * j * angular_velocity), sin(i * j * angular_velocity));
       auto window_element = window_data ? * (window_data + j) : 1;
-      auto element = *(X_data + j*X_stride) * window_element;
+      auto x = (j < number_of_samples) ? *(X_data + j * X_stride) : 0;
```

Comment:
it looks like number_of_samples was replaced with dft_length almost everywhere except a few spots - are the spots where it's still being used correct? I haven't dug into the math, just asking.

Reply:
Yeah, a condition needed to be added for when the dft_length > the actual number of samples available in the tensor